### PR TITLE
(#1222) Fixed test methods sharing issue

### PR DESCRIPTION
**Pull request by fanifieiev**
This pull request is for issue [1222](https://github.com/yegor256/cactoos/issues/1222)

According to the article https://www.yegor256.com/2016/05/03/test-methods-must-share-nothing.html
the test methods must not share any methods or data.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1225](https://github.com/yegor256/cactoos/issues/1225) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1225](https://github.com/yegor256/cactoos/issues/1225) is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/C63314D6Z/ee990a08-c27e-40f6-bd09-86a7a4cd5d84); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/test/java/org/cactoos/iterator/IteratorOfTest.java [hunk 1]
```diff
@@ -60,7 +62,12 @@ public void emptyIteratorThrowsException() {
 
     @Test
     public void nonEmptyIteratorDoesNotHaveNext() {
-        final IteratorOf<Integer> iterator = this.iteratorWithFetchedElements();
+        final Iterator<Integer> iterator = new Skipped<>(
```
Comment (victornoel) [inline src/test/java/org/cactoos/iterator/IteratorOfTest.java]: @fanifieiev please do not rely on `Skipped` here, it's too far away from the tested class to make the test robust

File: src/test/java/org/cactoos/iterator/IteratorOfTest.java [hunk 2]
```diff
@@ -70,9 +77,15 @@ public void nonEmptyIteratorDoesNotHaveNext() {
 
     @Test
     public void nonEmptyIteratorThrowsException() {
+        final Iterator<Character> iterator = new Skipped<>(
```
Comment (victornoel) [inline src/test/java/org/cactoos/iterator/IteratorOfTest.java]: @fanifieiev please do not rely on `Skipped` here, it's too far away from the tested class to make the test robust

Comment (fanifieiev) [reply src/test/java/org/cactoos/iterator/IteratorOfTest.java]: @victornoel Agree, made the fix

**Comment by fanifieiev** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/cactoos/issues/1225#issuecomment-550396333))

@fanifieiev This is what I know about this job in [C63314D6Z](https://www.0crat.com/p/C63314D6Z), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#1225](https://github.com/yegor256/cactoos/issues/1225) is [in scope](http://www.zerocracy.com/policy.html#14) for 2days
  * The role is `REV`
  * The job is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel) for [2days](http://www.zerocracy.com/policy.html#8)
  * There is no monetary reward attached, it's a [free](http://www.zerocracy.com/policy.html#2) job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @fanifieiev[/z](https://www.0crat.com/u/fanifieiev): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C63314D6Z?q=%7Bjob%3A%27gh%3Ayegor256%2Fcactoos%231225%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by victornoel** [comment]
@paulodamaso it's good to merge

**Comment by paulodamaso** [comment]
@victornoel Thanks

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by 0crat** [comment]
The job [#1225](https://github.com/yegor256/cactoos/issues/1225) is now out of scope

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by sereshqua** [comment]
@victornoel please make sure you will try to find at least 3 issues during next CR, thanks

**Comment by victornoel** [comment]
@sereshqua yes

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @victornoel[/z](https://www.0crat.com/u/victornoel)

**Pull request merged**